lib/internal/bootstrap/node.js

```diff
@@ -156,13 +156,31 @@ const rawMethods = internalBinding('process_methods');
   process._getActiveHandles = rawMethods._getActiveHandles;
 
   process.getActiveResourcesInfo = function() {
+    let timeoutCount = 0;
+    const lists = ObjectValues(internalTimers.timerListMap);
+    for (let i = 0; i < lists.length; ++i) {
+      const list = lists[i];
+      let timer = list._idlePrev;
+      while (timer !== list) {
+        timeoutCount += timer.hasRef();
+        timer = timer._idlePrev;
+      }
+    }
+
+    let immediateCount = 0;
+    const queue = internalTimers.outstandingQueue.head !== null ?
```

Comment:
You actually want to iterate over both the queues here. `outstandingQueue` contains any `Immediate` objects that weren't handled during the last run of `processImmediate` due to an unhandled exception but `immediateQueue` could at the same time contain newly scheduled `Immediate` objects (destined to run in loop+1 cycle). Both would be considered as keeping the process alive.